pkg/controllers/management/nodepool/nodepool.go

```diff
@@ -319,3 +349,44 @@ func (c *Controller) updateNodeRoles(existing *v3.Node, nodePool *v3.NodePool, s
 
 	return c.Nodes.Update(toUpdate)
 }
+
+// requeue checks every 5 seconds if the node is still unreachable
+func (c *Controller) requeue(timeout time.Duration, np *v3.NodePool, node *v3.Node) {
+	t := getTaint(node.Spec.InternalNodeSpec.Taints, &unReachableTaint)
+	for t != nil {
+		time.Sleep(5 * time.Second)
+		exist, err := c.NodeLister.Get(node.Namespace, node.Name)
+		if err != nil {
+			c.syncmap[node.Name] = false
```

Comment:
This can be be moved outside the for loop and just break here, default logic would to always be kill the map entry and return when exiting the go-routine